### ISSUE TITLE
Make field action_id an array

### ISSUE THREAD

**Issue by gubser** [mami-project/pto-core#6]
One desired property of the observatory is that a query which has been executed in the past should return the exact same results when reexecuted later.
This implies that the observatory should know which observations are existing and when they are valid or invalid _for every point in time_.

The solution to this is that the action_id field is an array of the following form:

```
action_ids: [{id: <int>, valid: <bool>}, (...), ...]
```

**Example:**

```
action_ids: [{id: 117, valid: true}, {id: 67, valid: false}, {id: 64, valid: true}]
```

The observation was created in 64, then invalidated in 67 and validated again in 117.
This means the observation:
- didn't exist before 64
- was valid from 64 to 66 and from 117 up
- was invalid from 67 to 116

**MongoDB**
The field `valid` can be removed when we make sure that the first element is always the most recent. The query

```
db.find({..., `valid`: true})
```

can be replaced with

```
db.find({..., `action_ids.0.valid`: true})
```

**Comment by gubser** [comment]
actually call it `action_ids` because it's an array and not an integer (as the case in metadata database)

**Comment by gubser** [comment]
implemented in 41b8292d3285046b49c9838dbf8ab2343474be08

**Comment by gubser** [comment]
finished in 988f91bb3713f2dc02cc50931565166527e2b3db
